Classes/Issues/Merge/IssueMergeSectionController.swift

```diff
@@ -65,13 +65,28 @@ MergeButtonDelegate {
         var viewModels = [ListDiffable]()
 
         if object.contexts.count > 0 {
-            let success = object.contexts.reduce(true, { $0 && $1.state == .success })
-            viewModels.append(IssueMergeSummaryModel(
-                title: success ?
-                    NSLocalizedString("All checks passed", comment: "")
-                    : NSLocalizedString("Some checks failed", comment: ""),
-                state: success ? .success : .failure
-            ))
+            let states = object.contexts.map { $0.state }
+            let state: IssueMergeSummaryModel.State
+            let stateDescription: String
+            let failureDescription = NSLocalizedString("Some checks failed", comment: "")
+            let containsFailure = states.contains(.failure) || states.contains(.error)
+            switch states {
+            case _ where containsFailure:
+                state = .failure
+                stateDescription = failureDescription
+            case let states where states.contains(.pending) && !containsFailure:
```

Comment:
`!containsFailure` will get hit by the previous case

Reply:
That's the inverse (`containsFailure`) right? Or am I missing something?

Reply:
Good catch!